# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1484008891 28800
#      Mon Jan 09 16:41:31 2017 -0800
# Branch stable
# Node ID f2456a17490c9f35a54fd4674eb6118c641635a2
# Parent  fdf9f98c3b728b74250d9e840b1a505b68b9c43a
compare_plot_demos: fix spurious link to next page

* html_compare_plot_demos.m: Fix off-by-one error that generates a bad link
to a next page when the total number of demos divides evenly by the number of
images per page.

diff --git a/scripts/testfun/private/html_compare_plot_demos.m b/scripts/testfun/private/html_compare_plot_demos.m
--- a/scripts/testfun/private/html_compare_plot_demos.m
+++ b/scripts/testfun/private/html_compare_plot_demos.m
@@ -83,17 +83,17 @@ function html_compare_plot_demos (toolki
   n = findstr (template, anchor);
   header = strtrim (template(1:n-1));
   trailer = strtrim (template(n+numel(anchor):end));
 
   page = 1;
   do
     start_fig = (page - 1) * in.plots_per_page + 1;
     stop_fig = page * in.plots_per_page;
-    last_page = stop_fig > numel (in.figfiles);
+    last_page = stop_fig >= numel (in.figfiles);
     if (last_page)
       stop_fig = numel (in.figfiles);
     endif
 
     fid = fopen (in.output_fmt (page), "w");
     unwind_protect
       fprintf (fid, "%s\n", header);
       fprintf (fid, "<h2>Generated on %s by %s with GNU Octave %s</h2>\n", ...
